packages/js/src/core/sign-in.ts

```diff
@@ -0,0 +1,47 @@
+import qs from 'query-string';
+
+import { withDefaultScopes } from '../utils/scope';
+
+export type SignInUriParameters = {
+  authorizationEndpoint: string;
+  clientId: string;
+  redirectUri: string;
+  codeChallenge: string;
+  state: string;
+  scope?: string | string[];
+  resource?: string[];
+};
+
+enum ResponseType {
+  code = 'code',
+}
```

Comment:
Do we have other response type options in the future?

Reply:
Yes, add when needed. prefer not to pre-define.